src/jemalloc.c

```diff
@@ -2745,6 +2770,14 @@ bool free_fastpath(void *ptr, size_t size, bool size_hint) {
 
 	tcache_t *tcache = tsd_tcachep_get(tsd);
 	cache_bin_t *bin = tcache_small_bin_get(tcache, alloc_ctx.szind);
+
+	/*
+	 * We "should" junk here, if set to.  But junking being enabled means
+	 * that we're on the fast path, which means we failed our fast-path
+	 * check above.
+	 */
+	assert(!opt_junk_free);
```

Comment:
Hello yinan,
from your id, i just curious are you from VMW*** company?  :)
actually i also quite interested in jemalloc, I had written some learning note in 2018, it can be accessed from https://github.com/everschen/tools/blob/master/DOC/Jemalloc.pdf, 
especially about the unit test architecture, it attracted me quite a lot.
maybe later this year, i would like to join you to continue to learn the unit test.

Reply:
Reworded.